src/output-json.c

```diff
@@ -839,11 +840,21 @@ JsonBuilder *CreateEveHeader(const Packet *p, enum OutputJsonLogDirection dir,
         JsonAddrInfoInit(p, dir, &addr_info);
         addr = &addr_info;
     }
-    jb_set_string(js, "src_ip", addr->src_ip);
-    jb_set_uint(js, "src_port", addr->sp);
-    jb_set_string(js, "dest_ip", addr->dst_ip);
-    jb_set_uint(js, "dest_port", addr->dp);
-    jb_set_string(js, "proto", addr->proto);
+    if (strlen(addr->src_ip) > 0) {
+        jb_set_string(js, "src_ip", addr->src_ip);
+    }
```

Comment:
Should we worry about scanning the string on almost every log entry? I think we could just check for the first byte being != '\0'?

Reply:
I think we can count on `strlen` being optimized by the library.

Reply:
> I think we can count on strlen being optimized by the library.

I would not bet on that